zerver/lib/streams.py

```diff
@@ -246,6 +246,9 @@ def list_to_streams(streams_raw: Iterable[Mapping[str, Any]],
         stream_name = stream_dict["name"]
         stream = existing_stream_map.get(stream_name.lower())
         if stream is None:
+            if autocreate and not user_profile.is_realm_admin and \
```

Comment:
@timabbott This checks seems straightforward enough, but I wonder if we want to do it higher up the stack?  It's possible that this really is the best place to put this check in terms of code simplicity, so I don't have a strong opinion here.

Reply:
Apparently we do similar checks for guest users in the function, so you can probably disregard my previous comment here.  (We discussed it a bit on chat.)

Reply:
This belongs in the `autocreate=True` code path a bit further down.

Reply:
So do I also put this check down with the others? I did add `if autocreate` here though.

Reply:
yes.